### ISSUE TITLE
Mocha/minitest issue in test_helper.rb

### ISSUE THREAD

**Issue by Marinlemaignan** [lynndylanhurley/devise_token_auth#1149]
Depending on which versions bundler would have installed it can happen to run into [mocha#324](https://github.com/freerange/mocha/issues/324)
we would need to investigate that as to make this library as simple as possible

a quick fix would be 
```
begin
  require 'mocha/minitest
rescue
  require 'mocha/mini_test'
end
```

**Comment by Evan-M** [comment]
It is probably better to add min and max versions of mocha to the `.gemspec` file.
It looks like `1.3.0` only has `mocha/mini_test.rb` whereas `1.4.0` and `1.5.0` switch to the updated `mocha/minitest.rb` with a deprecation warning added to the older `mocha/mini_test.rb`.

If you really want to wrap it in a `begin`/`rescue` block, you should probably rescue the `LoadError` exception explicitly, and check that the exception was in fact from attempting to load the missing version of `mocha/minitest`:
~~~ruby
begin
  require 'mocha/minitest'
rescue LoadError => e
  raise unless e.message =~ /minitest/
  require 'mocha/mini_test'
end

**Comment by Evan-M** [comment]
I wanted to add here that the `mocha` dependency was added way back in #323 (albeit incorrectly).  I reported the incorrect usage in #1137, and resolved it with #1139.

That said, as far as I can tell, the only place where `mocha` is actually used within the test suite is in `devise_token_auth/test/controllers/devise_token_auth/omniauth_callbacks_controller_test.rb` (and hence why it was included in that test file only, as opposed to the whole test suite).

Instead of wrapping the `require` in a `begin`/`rescue` block (and attempting to cleanly support both the older and newer versions of `mocha`), I think a better approach here is to refactor the test in question and remove the dependency on mocha altogether.
